packages/client/webapp/src/utils/TA2/stratify.ts

```diff
@@ -0,0 +1,25 @@
+// Providing the ID of 3 Models (model A, model B, and the type Model)
+// Create a new model of based off of the stratification
+
+import { json } from 'd3';
+
+// TODO: Better error handling (models not found for example)
+//       Override function to also accept 2 typed models (need to chat with TA2 to fully understand)
+export async function stratify(modelA, modelB, typeModel) {
+	console.log('Start Stratify');
+
+	if (!modelA || !modelB || !typeModel) {
+		return json('An ID must be provided for each model');
```

Comment:
Since this function is running on the frontend, we don't need to wrap the response in JS Object Notation, we could just return the string or an object directly.

Also, since d3 is just the library we're using for rendering graphs we shouldn't need to use its general purpose JSON utility functions at any point.

I could see an argument for either throwing an error:

```suggestion
		throw new Error('An ID must be provided for each model');
```

or logging the error directly to the console and returning an empty object `{}` or something.

Not sure if there's an established precedent in the app for where and how we want to handle errors so either of those would be fine.

Reply:
I'm for throwing Errors in JS, but we have to make sure they are properly handle!
So for each `throw new Error` I would like to see a test handling it.

Reply:
Good points. I have written this test case down. Currently just testing the fail case